sdk/go/postgres/db_value.go

```diff
@@ -0,0 +1,269 @@
+}
+
+func (n DbValue) GetUint8() uint8 {
+	if g, w := n.Kind(), DbValueKindUint8; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.(uint8)
+}
+
+func (n *DbValue) SetUint8(v uint8) {
+	n.val = v
+	n.kind = DbValueKindUint8
+}
+
+func DbValueUint16(v uint16) DbValue {
+	return DbValue{kind: DbValueKindUint16, val: v}
+}
+
+func (n DbValue) GetUint16() uint16 {
+	if g, w := n.Kind(), DbValueKindUint16; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.(uint16)
+}
+
+func (n *DbValue) SetUint16(v uint16) {
+	n.val = v
+	n.kind = DbValueKindUint16
+}
+
+func DbValueUint32(v uint32) DbValue {
+	return DbValue{kind: DbValueKindUint32, val: v}
+}
+
+func (n DbValue) GetUint32() uint32 {
+	if g, w := n.Kind(), DbValueKindUint32; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.(uint32)
+}
+
+func (n *DbValue) SetUint32(v uint32) {
+	n.val = v
+	n.kind = DbValueKindUint32
+}
+
+func DbValueUint64(v uint64) DbValue {
+	return DbValue{kind: DbValueKindUint64, val: v}
+}
+
+func (n DbValue) GetUint64() uint64 {
+	if g, w := n.Kind(), DbValueKindUint64; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.(uint64)
+}
+
+func (n *DbValue) SetUint64(v uint64) {
+	n.val = v
+	n.kind = DbValueKindUint64
+}
+
+func DbValueFloating32(v float32) DbValue {
+	return DbValue{kind: DbValueKindFloating32, val: v}
+}
+
+func (n DbValue) GetFloating32() float32 {
+	if g, w := n.Kind(), DbValueKindFloating32; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.(float32)
+}
+
+func (n *DbValue) SetFloating32(v float32) {
+	n.val = v
+	n.kind = DbValueKindFloating32
+}
+
+func DbValueFloating64(v float64) DbValue {
+	return DbValue{kind: DbValueKindFloating64, val: v}
+}
+
+func (n DbValue) GetFloating64() float64 {
+	if g, w := n.Kind(), DbValueKindFloating64; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.(float64)
+}
+
+func (n *DbValue) SetFloating64(v float64) {
+	n.val = v
+	n.kind = DbValueKindFloating64
+}
+
+func DbValueStr(v string) DbValue {
+	return DbValue{kind: DbValueKindStr, val: v}
+}
+
+func (n DbValue) GetStr() string {
+	if g, w := n.Kind(), DbValueKindStr; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.(string)
+}
+
+func (n *DbValue) SetStr(v string) {
+	n.val = v
+	n.kind = DbValueKindStr
+}
+
+func DbValueBinary(v []uint8) DbValue {
+	return DbValue{kind: DbValueKindBinary, val: v}
+}
+
+func (n DbValue) GetBinary() []uint8 {
+	if g, w := n.Kind(), DbValueKindBinary; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
+	}
+	return n.val.([]uint8)
+}
+
+func (n *DbValue) SetBinary(v []uint8) {
+	n.val = v
+	n.kind = DbValueKindBinary
+}
+
+func DbValueDbNull() DbValue {
```

Comment:
Should we have something like an `IsNull` function?  What sort of experience should we expect for loading a value from a NULLable column, especially if the developer wants graceful failure rather than panic?

(Again this is me being ignorant of How Things Are in Go.)

Reply:
Let me clarify what I'm driving at.  Consider a People table with two (interesting) columns, Age (integer) and MiddleName (string).  Both are NULLable in the database (to reflect unknown age and people who don't have a middle name).

I want to query for the set of People and process that list, for example rendering to HTML (printing `unknown` for NULL ages and `-` for no middle name), or marshalling to JSON (where I guess I'd omit the NULL fields), but for generality going via some intermediate Go structure.

How (generally) do I do this?

If memory serves, for the intermediate structure to allow `nil`, its values would have to be pointers e.g. `{ Age *int, MiddleName *string }`.  Is that right?  If so, what is the outline for populating them, something like:

```
var age
if v.Kind() == DbNull {
  age = nil
} else {
  age = &v.GetInt()
}

person := Person { Age: age, ... }
```

Is that right (modulo Go syntax!), and if so, does it seem like what a Go programmer would expect to right?

To be clear, I am _not_ suggesting it's not!  I'm not familiar with Go idioms - for all I know this is _exactly_ what a Go programmer would find familiar.  I'm just confirming that we are happy with whatever the pattern for nullable columns is.

Hope this makes more sense now - sorry for the somewhat muddled question first time around!

Reply:
> If memory serves, for the intermediate structure to allow nil, its values would have to be pointers e.g. { Age *int, MiddleName *string }. Is that right?

I do believe that is right, yes. When I created a struct without pointers, the struct is initialized with default values (i.e. 0 for int, "" for string, etc.)